scalafix-core/shared/src/main/scala/scalafix/internal/util/PrettyType.scala

```diff
@@ -273,66 +286,54 @@ class PrettyType private (table: SymbolTable, shorten: QualifyStrategy) {
                 )
               )
             case k.CLASS =>
-              val decls = declarations.map(this.info)
-              val primaryConstructor = decls.collectFirst {
-                case i if i.kind.isConstructor && i.is(p.PRIMARY) =>
-                  i
-              }
-              val ctor = primaryConstructor match {
-                case Some(i) =>
-                  toStat(i) match {
-                    case Some(d: Decl.Def) =>
-                      val paramss = d.paramss match {
-                        case Nil :: Nil if !info.is(p.CASE) => Nil
-                        case els => els
-                      }
-
-                      Ctor.Primary(
-                        toMods(i),
-                        Name.Anonymous(),
-                        paramss
-                      )
-                    case Some(els) =>
-                      fail(els)
-                    case None =>
-                      Ctor.Primary(Nil, Name(""), Nil)
-                  }
-                case _ =>
+              val ctor: Ctor.Primary = declarations
+                .collectFirst {
+                  case i if i.kind.isConstructor && i.is(p.PRIMARY) =>
+                    toTree(i) match {
+                      case ctor @ Ctor.Primary(_, _, Nil :: Nil)
+                          if !info.is(p.CASE) =>
+                        ctor.copy(paramss = Nil)
```

Comment:
Why is this necessary?

Reply:
To produce the syntax `class Foo` instead of `class Foo()`. I added a comment to document the reason.